### PR TITLE
Feature/dfe 1060 derive latest release

### PR DESCRIPTION
**Pull request by shakes90**

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/ReleasesController.cs [hunk 1]
```diff
@@ -159,5 +160,11 @@ private bool ReleaseExists(Guid id)
         {
             return _context.Releases.Any(e => e.Id == id);
         }
+
+        private int NextOrder(Release release)
+        {
+            var rel = _context.Releases.OrderByDescending(r => r.Order).FirstOrDefault(r => r.Publication.Id == release.PublicationId);
```
Comment (richardpopplehive) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/ReleasesController.cs]: Does the FirstOrDefault not return a default value? I.e. do you need the next line?

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 2]
```diff
@@ -34,45 +37,33 @@ public List<Publication> List()
...
+                .Include(p => p.Releases)
+                .Include(p => p.Methodologies)
+                .ToList();
+            
+            return mapper.Map<List<PublicationViewModel>>(publications);
+        }
+        
+        private bool IsLatestRelease(Release release)
+        {
+            return !release.Publication.Releases.Exists(
```
Comment (richardpopplehive) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: Is there any advantage of sticking this on the Release object itself as a transient field?

File: src/GovUk.Education.ExploreEducationStatistics.Content.Api/Migrations/20190718135116_AddOrderFieldToRelease.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+namespace GovUk.Education.ExploreEducationStatistics.Content.Api.Migrations
+{
+    public partial class AddOrderFieldToRelease : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "Order",
+                table: "Releases",
+                nullable: false,
```
Comment (richardpopplehive) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Api/Migrations/20190718135116_AddOrderFieldToRelease.cs]: Are we happy that all of the data in the dev database is set to 0? I assume that actually would be okay?

Comment (shakes90) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Api/Migrations/20190718135116_AddOrderFieldToRelease.cs]: yep - it'll be OK as will get added as zero & only one release at the mo.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/ReleasesController.cs (hunk 1 above)
Comment (shakes90) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/ReleasesController.cs]: Returns null - there's no orelse as per java sadly.

**Pull request merged**